sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java

```diff
@@ -130,7 +130,7 @@ protected FileBasedSource(
    *
    * @throws IllegalArgumentException if this source is in {@link Mode#FILEPATTERN} mode.
    */
-  protected final MatchResult.Metadata getSingleFileMetadata() {
+  public final MatchResult.Metadata getSingleFileMetadata() {
```

Comment:
@tvalentyn what do you think about this and the FileIO changes being done in a different commit / pr . As its a change that is potentially useful to other sources api dev and maybe easier to reason about then if tied to this larger commit?

Reply:
SGTM, different commit should be sufficient.